botwo/botwo.py

```diff
@@ -0,0 +1,200 @@
+import fnmatch
+import botocore
+
+COPY_METHODS = {"copy", "copy_object", "copy_upload_part"}
+LIST_METHODS = {"list_objects", "list_objects_v2", "list_object_version"}
+
+
+class PaginatorWrapper(object):
+    """ This class wraps boto paginator """
+
+    def __init__(self, mapping, config, kwargs):
+        self.mapping = mapping
+        self.config = config
+        self.paginators = dict()
+        for client in self.mapping:
+            self.paginators[client] = self.mapping[client].get_paginator(kwargs["operation_name"])
+        self.default = self.paginators.get("default")
+
+    def paginate(self, **kwargs):
+        for profile in self.config:
+            mapping = self.config[profile]
+
+            if fnmatch.fnmatch(kwargs["Bucket"], mapping.get("source_bucket_pattern")):
```

Comment:
Again, if we declared `paginate(self, Bucket, **kwargs)`, would it work?  Would this become easier to read?